beacon-chain/blockchain/head.go

```diff
@@ -108,6 +112,17 @@ func (s *Service) saveHead(ctx context.Context, headRoot [32]byte) error {
 		return errors.Wrap(err, "could not save head root in DB")
 	}
 
+	// A chain re-org occurred, so we fire an event notifying the rest of the services.
+	if newHeadState.Slot() < currentHeadSlot {
```

Comment:
Not sure this is fully correct, while this would be one way to determine a re-org the correct way would be to check whether the incoming block is a direct child of the current head. If it's not then we can mark it as a re-org.

You could have re-orgs also when 
```go
newHeadState.Slot() > currentHeadSlot 
``` 
where the new head isn't a descendant of the current chain head. 
cc @terencechain

Reply:
Yeah my bad on this. I was only thinking reorg that affect assignments so it's not complete. Nishant is right. We can have the following:
100 - 101 - 103
      \- 102

Head can change from 101 to 102 to 103.  From 102 to 103 it's a reorg but slot isn't right.
Checking the direct child sounds good to me